src/components/organisms/EntityDetails/EntityDetailsContent/Settings/SettingsScheduling/Schedule.tsx

```diff
@@ -0,0 +1,189 @@
+
+const namingMap: {[key in Entity]: string} = {
+  'test-suites': 'test suite',
+  tests: 'test',
+};
+
+const quickOptions = [
+  {
+    label: 'Every 5 minutes',
+    value: '*/5 * * * *',
+  },
+  {
+    label: 'Every 30 minutes',
+    value: '*/30 * * * *',
+  },
+  {
+    label: 'Every hour',
+    value: '0 * * * *',
+  },
+  {
+    label: 'Every day',
+    value: '0 0 * * *',
+  },
+  {
+    label: 'Every week',
+    value: '0 0 * * 0',
+  },
+  {
+    label: 'Every month',
+    value: '0 0 1 * *',
+  },
+];
+
+const defaultCronString = '* * * * *';
+
+const Schedule: React.FC = () => {
+  const {entity, entityDetails} = useContext(EntityDetailsContext);
+
+  const [updateTest] = useUpdateTestMutation();
+  const [updateTestSuite] = useUpdateTestSuiteMutation();
+
+  const updateRequestsMap: {[key in Entity]: any} = {
+    'test-suites': updateTestSuite,
+    tests: updateTest,
+  };
+  const name = entityDetails?.name;
+  const schedule = entityDetails?.schedule;
+
+  const [cronString, setCronString] = useState(schedule || defaultCronString);
+  const [wasTouched, setWasTouched] = useState(false);
+  const [isValidFormat, setIsValidFormat] = useState(true);
+
+  if (!entity || !entityDetails) {
+    return null;
+  }
+
+  const onSave = () => {
+    updateRequestsMap[entity]({
+      id: name,
+      data: {
+        ...entityDetails,
+        schedule: cronString,
+      },
+    })
+      .then((res: any) => {
+        displayDefaultNotificationFlow(res, () => {
+          notificationCall('passed', `${uppercaseFirstSymbol(namingMap[entity])} schedule was successfully updated.`);
+        });
+        setWasTouched(false);
+      })
+      .catch((err: any) => {
+        displayDefaultErrorNotification(err);
+      });
+  };
+
+  const onCancel = () => {
+    setWasTouched(false);
+    setCronString(schedule || defaultCronString);
+  };
+
+  const onCronInput = (value: string, position: number) => {
+    const split = cronString.split(' ');
+    setCronString([...split.slice(0, position), value, ...split.slice(position + 1)].join(' '));
+    setWasTouched(true);
+  };
+
+  const getNextExecution = () => {
+    try {
+      const nextExecution = parser.parseExpression(cronString).next().toDate();
+      const duration = intervalToDuration({start: new Date(), end: nextExecution});
+
+      if (!isValidFormat) {
+        setIsValidFormat(true);
+      }
+      return `in ${formatDuration(duration)}`;
+    } catch (e) {
+      if (isValidFormat) {
+        setIsValidFormat(false);
+      }
+      return 'Invalid cron format';
+    }
+  };
+
+  const [minute, hour, day, month, dayOfWeek] = cronString.split(' ');
+
+  return (
+    <ConfigurationCard
+      title="Schedule"
+      description={`You can add a cronjob like schedule for your ${namingMap[entity]} which will then be executed automatically.`}
+      onConfirm={onSave}
+      onCancel={onCancel}
+      isButtonsDisabled={!wasTouched}
```

Comment:
When `!isValidFormat` it should rather disable the "Save" button - at the moment it's calling API with an invalid schedule

Reply:
User actually can successfully save invalid schedule format. It will just not work. So I actually decided this to be a non-blocking error. We tend to do it in several places in project